vizro-core/tests/unit/vizro/models/_controls/test_filter.py

```diff
@@ -111,6 +224,16 @@ def test_set_slider_values_incompatible_column_type(self, test_input, managers_o
         ):
             filter.pre_build()
 
+    def test_set_datepicker_values_incompatible_column_type(self, managers_one_page_two_graphs):
+        filter = vm.Filter(column="country", selector=vm.DatePicker())
+        model_manager["test_page"].controls = [filter]
+        with pytest.raises(
+            ValueError,
+            match=f"Chosen selector {vm.DatePicker().type} "
+            f"is not compatible with categorical column '{filter.column}'.",
```

Comment:
Shall we just test for both, numerical and temporal, and check for a little broader error message?

Reply:
Good spot Max! I tried to refactor some tests for testing `column_type` <-> `selector` (in)compatibility.  

@huong-li-nguyen @antonymilne can you also help us to double check the desired requirements. 
Are the following statements correct? 
If yes, then we have a bug because currently we allow to set a `categorical selector` for `numerical/temporal column_type`.

1. column_type is `numerical`
  1.1. allowed selectors: (RangeSlider, Slider)
  1.2. disallowed selectors: (Dropdown, Checklist, RadioItems, DatePicker)
2. column_type is `temporal`
  2.1. allowed selectors: (DatePicker)
  2.2. disallowed selectors: (Dropdown, Checklist, RadioItems, RangeSlider, Slider)
3. column_type is `categorical` 
  3.1. allowed selectors: (Dropdown, Checklist, RadioItems)
  3.2. disallowed selectors: (RangeSlider, Slider, DatePicker)

Reply:
1 and 3 seem right.

For 2) What happens if you add temporal data into the RangeSlider or Slider? If the underlying dash component can visualise it, then I would allow it as a selector. If not, then not.

So generally, the disallowed selectors are the ones where the underlying dash component cannot handle that data type and therefore does not visualise it properly.

Reply:
I think this is probably too restrictive. Just because something is numerical doesn't mean it's continuous or that every value in the range is valid. Hence someone might wish to use a categorical selector for a numerical column.

e.g. I want a form component for someone to choose their favourite prime number out of 6, 28, 496, 8128. It would make more sense for this to be a categorical selector like `RadioItems` rather than a slider.

I think the requirements should be:
1. column_type is `numerical`
  1.1. allowed selectors: (RangeSlider, Slider, Dropdown, Checklist, RadioItems)
  1.2. disallowed selectors: (DatePicker)
2. column_type is `temporal`
  2.1. allowed selectors: (DatePicker, Dropdown, Checklist, RadioItems)
  2.2. disallowed selectors: (RangeSlider, Slider) - but allowed if these work ok for dates actually
3. column_type is `categorical` 
  3.1. allowed selectors: (Dropdown, Checklist, RadioItems)
  3.2. disallowed selectors: (RangeSlider, Slider, DatePicker)

Looked at another way:
* the categorical selectors (Checklist, Dropdown, RadioItems) should work for everything because you can treat _any_ data with finite number of values as categorical if you really want to.
* the numerical selector (RangeSlider, Slider) only makes sense for data that can be ordered (ordinal). _Generally speaking_ this is numbers and dates, but in theory it could be something else e.g. T-shirts sizes XS, S, M, L, XL
* the date selector DatePicker only makes sense for dates and nothing else

Currently the only restriction we have (pre-datepicker) is that numerical selector (RangeSlider, Slider) only works for numerical data. This is already slightly overly restrictive (due to T-shirt size example) but that's ok because it's an edge case that no one really cares about.

Basically we should let people use whatever selector they like _so long as the underlying functionality works ok_.  Our code for numerical selector relies on finding the `min` and `max` of a column, and for ordinal data like In the T-shirt size example this is tricky to do in pandas, so it's fine that we don't support it. Think of it as like duck typing: a selector has certain requirements that the data must obey, e.g. it defines a `min` or it has a finite number of values. So long as these requirements are met it's ok to use that sort of data in that selector.

Reply:
Thanks for the input! I solved it, and it looks much cleaner to me now. What do you think? https://github.com/mckinsey/vizro/pull/319/commits/8f5a011b0a0e54fcac67552aab4941e4342d3a1b